pemFioi/quickAlgo/blockly_interface.js

```diff
@@ -491,6 +491,15 @@ function getBlocklyInterface(maxBlocks, nbTestCases) {
                this.mainContext.savePrograms(xml);
             }
 
+            // TODO: Maybe later remove the if and replace it with "can edit subject", because we might want to save
+            // the subject also.
+            if (saveSubject) {
+               var subjectNode = goog.dom.createElement("subject");
```

Comment:
blockly_interface doesn't manage the title and description modification, it hence shouldn't handle processing the data to be saved.

Instead of having "saveSubject" as argument to the function, it should call a function (from quickAlgo interface) to get all the data it needs.

Also, we'll talk about it, but I think we should move away from saving everything as proper XML.